### PR TITLE
Add a few additional failures to our notes doc

### PR DESCRIPTION
**Pull request by RoriCremer**
no automated testing needed--just documentation edits

### CODE REVIEW

File: scripts/variantstore/beta_docs/gvs-troubleshooting.md [hunk 1]
```diff
@@ -9,18 +9,32 @@ Generally, if you have started the GVS workflow and it failed after ingestion, o
...
+1. BulkIngestGenomes/GvsBulkIngestGenomes/hash/call-ImportGenomes/GvsImportGenomes/hash/call-GetUningestedSampleIds/gvs_ids.csv Required file output '/cromwell_root/gvs_ids.csv' does not exist.
    1. If you've attempted to run GVS more than once in the same BigQuery dataset, you may see this error. Please delete the dataset and create a new one. We recommend naming the new dataset something different than the one you deleted.
-4. AssignIds failure with error message: `BigQuery error in mk operation: Not found: Dataset`
-   1. This is saying that GVS was unable to find the BigQuery dataset specified in the inputs. If you haven't created a BigQuery dataset prior to running the workflow, you can follow the steps in [the quickstart](./gvs-quickstart.md). If you created it and still see this error, check the naming of the dataset matches your input specified and that the google project in the inputs is correct. Lastly, confirm you have set up the correct permissions for your Terra proxy account following the instructions in the quickstart. 
+1. AssignIds failure with error message: BigQuery error in mk operation: Not found: Dataset
```
Comment (rsasch) [inline scripts/variantstore/beta_docs/gvs-troubleshooting.md:21]:
```suggestion
1. AssignIds failure with error message: `BigQuery error in mk operation: Not found: Dataset`
```

File: scripts/variantstore/beta_docs/gvs-troubleshooting.md [hunk 2]
```diff
@@ -9,18 +9,32 @@ Generally, if you have started the GVS workflow and it failed after ingestion, o
...
+   1. This is saying that GVS was unable to find the BigQuery dataset specified in the inputs. If you haven't created a BigQuery dataset prior to running the workflow, you can follow the steps in the quickstart. If you created it and still see this error, check the naming of the dataset matches your input specified and that the google project in the inputs is correct. Lastly, confirm you have set up the correct permissions for your Terra proxy account following the instructions in the quickstart.
+1. Ingest failure with error message: raise ValueError("vcf column not in table")
```
Comment (rsasch) [inline scripts/variantstore/beta_docs/gvs-troubleshooting.md:23]:
```suggestion
1. Ingest failure with error message: `raise ValueError("vcf column not in table")`
```

File: scripts/variantstore/beta_docs/gvs-troubleshooting.md [hunk 3]
```diff
@@ -9,18 +9,32 @@ Generally, if you have started the GVS workflow and it failed after ingestion, o
...
+   1. You will want to completely start over and delete your BQ dataset--and then re-create it. It can have the exact same name.
+1. Ingest failure: There is already a list of sample names. This may need manual cleanup. Exiting.
+   1. Clean up the BQ dataset manually by deleting it and recreating it fresh
+   1. Make sure to keep the call caching on and run it again
+1. Ingest failure with error message: A USER ERROR has occurred: Cannot be missing required value for ___
+   1. (e.g. alternate_bases.AS_RAW_MQ, RAW_MQandDP or RAW_MQ)
+   1. This means that there is at least one incorrectly formatted sample in your data model. Confirm your GVCFs are reblocked. If the incorrectly formatted samples are a small portion of your callset and you wish to just ignore them, simply delete the from the data model and restart the workflow without them. There should be no issue with starting from here as none of these samples were loaded.
+1. Extract failure with OSError: Is a directory. If you point your extract to a directory that doesn’t already exist, it will not be happy about this. Simply make the directory and run the workflow again.
+1. Ingest failure with: Lock table error
```
Comment (rsasch) [inline scripts/variantstore/beta_docs/gvs-troubleshooting.md:38]:
```suggestion
1. Ingest failure with: `Lock table error`
```

File: scripts/variantstore/beta_docs/gvs-troubleshooting.md [hunk 4]
```diff
@@ -9,18 +9,32 @@ Generally, if you have started the GVS workflow and it failed after ingestion, o
...
 
-## Reblocking
-1. `htsjdk.tribble.TribbleException$MalformedFeatureFile: Unable to parse header with error: Your input file has a malformed header: We never saw the required CHROM header line (starting with one #) for the input VCF file, for input source: file:///cromwell_root/v1_[uuid]`
-   1. If you are running ReblockGVCF from a TDR snapshot, you will see this error if you did not check the “Convert DRS URLs to Google Cloud Storage Paths (gs://)" box before exporting the snapshot.
-2. GVS is running very slowly!
-   1. If your GVS workflow is running very slowly compared to the example runtimes in the workspace, you may have run GVS on GVCFs that have not been reblocked. Confirm your GVCFs are reblocked.
\ No newline at end of file
+1. It is important to verify that the data has ALL made it into the BQ dataset or not
```
Comment (rsasch) [inline scripts/variantstore/beta_docs/gvs-troubleshooting.md:40]: this seems incomplete

File: scripts/variantstore/beta_docs/gvs-troubleshooting.md (hunk 3 above)
Comment (rsasch) [reply scripts/variantstore/beta_docs/gvs-troubleshooting.md:38]: this seems incomplete; what is the user to do if they run into this error?

**Review by rsasch — CHANGES_REQUESTED** [review]
Some specific changes, also I think it would be useful to have a consistent way of attaching the failures to a specific workflow, sub-workflow and task for easier use.

**Comment by RoriCremer** [comment]
at some point in our notes we added `1. It is important to verify that the data has ALL made it into the BQ dataset or not
` but i'm not sure how to do this with Beta users or if it makes any sense at all

File: scripts/variantstore/beta_docs/gvs-troubleshooting.md [hunk 5]
```diff
@@ -8,19 +8,36 @@ Generally, if you have started the GVS workflow and it failed after ingestion, o
 2. `BadRequestException: 400 Bucket is a requester pays bucket but no user project provided.`
    1. GVS can ingest data from a requester pays bucket by setting the optional `billing_project_id` input variable. This variable takes a string of a Google project ID to charge for the egress of the GVCFs and index files.
 
-## Runtime errors
+## Ingestion-Specific Issues"
+1. GVS is running very slowly!
+   1. If your GVS workflow is running very slowly compared to the example runtimes in the workspace, you may have run GVS on GVCFs that have not been reblocked. Confirm your GVCFs are reblocked.
```
[the remark this answers is not in the document]
Comment (RoriCremer) [reply scripts/variantstore/beta_docs/gvs-troubleshooting.md:13]: well the goal was to find all the ones that I could!!!!

[the remark this answers is not in the document]
Comment (kbergin) [reply scripts/variantstore/beta_docs/gvs-troubleshooting.md:13]: Our documentation points to the reblocking workflow and documentation regularly and the workflow is in our GVS workspace. But you could add here "Confirm your GVCFs are reblocked by ..... "

File: scripts/variantstore/beta_docs/gvs-troubleshooting.md [hunk 6]
```diff
@@ -8,19 +8,36 @@ Generally, if you have started the GVS workflow and it failed after ingestion, o
 2. `BadRequestException: 400 Bucket is a requester pays bucket but no user project provided.`
    1. GVS can ingest data from a requester pays bucket by setting the optional `billing_project_id` input variable. This variable takes a string of a Google project ID to charge for the egress of the GVCFs and index files.
 
-## Runtime errors
+## Ingestion-Specific Issues"
```
Comment (kbergin) [inline scripts/variantstore/beta_docs/gvs-troubleshooting.md:11]:
```suggestion
### Ingestion-Specific Issues
```

I think you'd want this to be three headers in and the " seems to be a typo

File: scripts/variantstore/beta_docs/gvs-troubleshooting.md [hunk 7]
```diff
@@ -8,19 +8,36 @@ Generally, if you have started the GVS workflow and it failed after ingestion, o
...
+   1. If you've attempted to run GVS more than once in the same BigQuery dataset, you may see this error. Please delete the dataset and create a new one. We recommend naming the new dataset something different than the one you deleted.
+1. AssignIds failure with error message: `BigQuery error in mk operation: Not found: Dataset`
+   1. This is saying that GVS was unable to find the BigQuery dataset specified in the inputs. If you haven't created a BigQuery dataset prior to running the workflow, you can follow the steps in the quickstart. If you created it and still see this error, check the naming of the dataset matches your input specified and that the google project in the inputs is correct. Lastly, confirm you have set up the correct permissions for your Terra proxy account following the instructions in the quickstart.
+1. Ingest failure with error message: `raise ValueError("vcf column not in table")`
+   1. if you have given an incorrect name for the vcf column or the vcf index column
+   1. You can simply restart the workflow with the correct names
+1. Ingest failure with error message: `Invalid resource name projects/gvs_internal; Project id: gvs_internal.`
```
Comment (kbergin) [inline scripts/variantstore/beta_docs/gvs-troubleshooting.md:29]:
```suggestion
1. Ingest failure with error message: `Invalid resource name projects/{your_project_id}; Project id: {your_project_id}.`
```

File: scripts/variantstore/beta_docs/gvs-troubleshooting.md [hunk 8]
```diff
@@ -8,19 +8,36 @@ Generally, if you have started the GVS workflow and it failed after ingestion, o
...
+   1. This occurs if you have given the incorrect name of the project.
+   1. Restart the workflow with the correct name
+1. Ingest failure with `Max id is 0. Exiting.`
+   1. You will want to completely start over and delete your BQ dataset--and then re-create it. It can have the exact same name.
+1. Ingest failure: There is already a list of sample names. This may need manual cleanup. Exiting.
+   1. Clean up the BQ dataset manually by deleting it and recreating it fresh
+   1. Make sure to keep the call caching on and run it again
+1. Ingest failure with error message: `A USER ERROR has occurred: Cannot be missing required value for `___
```
Comment (kbergin) [inline scripts/variantstore/beta_docs/gvs-troubleshooting.md:37]: Is the error message really "Cannot be missing required value for" ? Just seems like an improperly formed sentence

File: scripts/variantstore/beta_docs/gvs-troubleshooting.md [hunk 9]
```diff
@@ -8,19 +8,36 @@ Generally, if you have started the GVS workflow and it failed after ingestion, o
...
+   1. (e.g. alternate_bases.AS_RAW_MQ, RAW_MQandDP or RAW_MQ)
+   1. This means that there is at least one incorrectly formatted sample in your data model. Confirm your GVCFs are reblocked. If the incorrectly formatted samples are a small portion of your callset and you wish to just ignore them, simply delete the from the data model and restart the workflow without them. There should be no issue with starting from here as none of these samples were loaded.
```
Comment (kbergin) [inline scripts/variantstore/beta_docs/gvs-troubleshooting.md:39]: You could link here that the full list of required information in a GVCF to work in GVS is here: 
https://github.com/broadinstitute/gatk/blob/ah_var_store/scripts/variantstore/beta_docs/run-your-own-samples.md#gvcf-annotations

File: scripts/variantstore/beta_docs/gvs-troubleshooting.md [hunk 10]
```diff
@@ -8,19 +8,36 @@ Generally, if you have started the GVS workflow and it failed after ingestion, o
...
+1. Extract failure with OSError: Is a directory. If you point your extract to a directory that doesn’t already exist, it will not be happy about this. Simply make the directory and run the workflow again.
```
Comment (kbergin) [inline scripts/variantstore/beta_docs/gvs-troubleshooting.md:40]:
```suggestion
1. Extract failure with OSError: Is a directory. 
      1. If you point your extract to a directory that doesn’t already exist, the workflow fails. Make the directory and run the workflow again.
```
I'm surprised this is an error the workflow can run into because our documentation suggests people get the workspace bucket and then add a subdirectory to that based on where they want the callset written. Usually that bucket doesn't actually already exist.. Am I misunderstanding your description of this error?

File: scripts/variantstore/beta_docs/gvs-troubleshooting.md [hunk 11]
```diff
@@ -8,19 +8,36 @@ Generally, if you have started the GVS workflow and it failed after ingestion, o
...
+1. Ingest failure with: `Lock table error`
+   1. This means that the lock table has been created, but that the ingest has failed soon after or that perhaps during manual cleanup from another failure, some underlying data was deleted
+   1. The lock table can simply be deleted -- `sample_id_assignment_lock` -- and the ingest can be kicked off again
```
Comment (kbergin) [inline scripts/variantstore/beta_docs/gvs-troubleshooting.md:43]: We don't talk about the lock table anywhere else in our documentation. I think this will need more detail to be clear to most people. Where is the lock table? How do they find it and delete it? If it picks up where ingest left off, let's reassure them of that here.

File: scripts/variantstore/beta_docs/gvs-troubleshooting.md [hunk 12]
```diff
@@ -8,19 +8,36 @@ Generally, if you have started the GVS workflow and it failed after ingestion, o
...
+   1. Clean up the BQ dataset manually by deleting it and recreating it fresh
+   1. Make sure to keep the call caching on and run it again
+1. Ingest failure with error message: `A USER ERROR has occurred: Cannot be missing required value for `___
+   1. (e.g. alternate_bases.AS_RAW_MQ, RAW_MQandDP or RAW_MQ)
+   1. This means that there is at least one incorrectly formatted sample in your data model. Confirm your GVCFs are reblocked. If the incorrectly formatted samples are a small portion of your callset and you wish to just ignore them, simply delete the from the data model and restart the workflow without them. There should be no issue with starting from here as none of these samples were loaded.
+1. Extract failure with OSError: Is a directory. If you point your extract to a directory that doesn’t already exist, it will not be happy about this. Simply make the directory and run the workflow again.
+1. Ingest failure with: `Lock table error`
+   1. This means that the lock table has been created, but that the ingest has failed soon after or that perhaps during manual cleanup from another failure, some underlying data was deleted
```
Comment (kbergin) [inline scripts/variantstore/beta_docs/gvs-troubleshooting.md:42]: Given the lock table hasn't been mentioned in any of our documentation, I'd reword this to something like

"This error is describing an issue found with one of the tables in the GVS BigQuery dataset called the 'lock table'. It indicates that the lock table has been created, but that the ingest failed soon after or that during manual cleanup from another failure, some underlying data was deleted and now the table is in a broken state."

**Review by kbergin — CHANGES_REQUESTED** [review]
I'm sorry this took me SO long to review. Comments throughout.

**Comment by kbergin** [comment]
Generally, could you add sections for failures in the other high level steps of GVS? Or if GVS fails in any step besides ingest does the user need to start over from scratch? It'd be great to indicate that ie "If your workflow fails in any step after ingest, unfortunately you will need to delete your BigQuery dataset and start from the beginning." ?

File: scripts/variantstore/beta_docs/gvs-troubleshooting.md [hunk 13]
```diff
@@ -8,19 +8,36 @@ Generally, if you have started the GVS workflow and it failed after ingestion, o
...
+   1. if you have given an incorrect name for the vcf column or the vcf index column
+   1. You can simply restart the workflow with the correct names
+1. Ingest failure with error message: `Invalid resource name projects/gvs_internal; Project id: gvs_internal.`
+   1. This occurs if you have given the incorrect name of the project.
+   1. Restart the workflow with the correct name
+1. Ingest failure with `Max id is 0. Exiting.`
+   1. You will want to completely start over and delete your BQ dataset--and then re-create it. It can have the exact same name.
+1. Ingest failure: There is already a list of sample names. This may need manual cleanup. Exiting.
+   1. Clean up the BQ dataset manually by deleting it and recreating it fresh
+   1. Make sure to keep the call caching on and run it again
```
Comment (kbergin) [inline scripts/variantstore/beta_docs/gvs-troubleshooting.md:36]: Looks like call caching specifically needs to be off for this to be successful, based on recent user error. I am going to bet that's true for both this error and the max id is 0 error.

File: scripts/variantstore/beta_docs/gvs-troubleshooting.md (hunk 6 above)
Comment (RoriCremer) [reply scripts/variantstore/beta_docs/gvs-troubleshooting.md:11]: should they all be 3 headers in then?

Comment (kbergin) [reply scripts/variantstore/beta_docs/gvs-troubleshooting.md:11]: Sorry, I misread this and thought it was a subheader under ##Runtime errors.. I think that actually is how it should be. 

##Runtime errors
### Ingestion errors
[insert one sentence or so explanation of which tasks are 'ingestion' here]
1. error one 
2. error two

### Troubleshooting other pipeline failures
1. Other errors in document
2.
3.

File: scripts/variantstore/beta_docs/gvs-troubleshooting.md (hunk 10 above)
Comment (RoriCremer) [reply scripts/variantstore/beta_docs/gvs-troubleshooting.md]: Wait -- this was just me reformatting something that was already there. Ok I'm gonna delete it, but I'm unhappy with this feedback

Comment (kbergin) [reply scripts/variantstore/beta_docs/gvs-troubleshooting.md]: I see that it was just a reformat, but the error description and remedy is still confusing to me given we regularly set gcs_extract_dir that do not exist prior to running the workflow and it doesn't fall over.. If you think it's an accurate description of the error then you can leave it!

**Review by kbergin — APPROVED** [review]
